freqtrade/persistence/models.py

```diff
@@ -861,6 +861,8 @@ def recalc_trade_from_orders(self):
                 total_stake += tmp_price * tmp_amount
 
         if total_amount > 0:
+            # TODO-lev: This should update leverage as well -
+            # as averaged trades might have different leverage
```

Comment:
Would this work?
```
for o in self.orders:
    total_leverage += o.leverage * o.filled/total_amount
```

Reply:
I guess ... 
the problem is - `Orders.leverage` is not filled and is always 1 - since ccxt does not return leverage as part of order responses so the leverage parsing always falls back to the column default.
to be clear, I don't think that's a ccxt problem (exchange response doesn't contain leverage either) - but a problem of how we assign the column.

i guess we'll have to simply add the appropriate leverage when creating the order ... 
so it would work here and in tests, but not in reality...

Reply:
>  so it would work here and in tests, but not in reality...

Is it a problem if it doesn't work in reality? That sounds like an issue

Reply:
> Is it a problem if it doesn't work in reality

you bet :laughing: 
it's also mentioned in #6224 - which points out that leverage in the orders table is always 1 (that's how i discovered it first, and can confirm that all my live databases have leverage=1 in the orders table, too).

so far, that's not a problem as it's a unused field - but with the above change, it wouldn't be, so we should have correct content in the field.

Reply:
@samgermain thinking about this, a much better question arises - is that even possible?
i buy 100$ worth of "COIN" with 5x leverage, and then again 200$ worth of coin at 3x leverage (so my leverage would be 4.5.

That's however not a valid leverage value - so i THINK what's going to happen is that the call to "3x leverage" will reduce my initial position to be at 3x leverage - so now i have 300$ at 3x leverage.

Reply:
> That's however not a valid leverage value -

It's valid on some exchanges

> That's however not a valid leverage value - so i THINK what's going to happen is that the call to "3x leverage" will reduce my initial position to be at 3x leverage - so now i have 300$ at 3x leverage.

On binance futures, in isolated mode, you can't reduce your leverage, if you increase a position for $100 at 5x leverage to 10x leverage, you now have a position for $50 at 10x leverage

In cross mode, if you decrease a position for $100 at 20x leverage to 10x leverage, you now have a position for $200 at 10x leverage

---------------------

Binance will also only have one leverage value for all trades with that currency, the same is true for gateio, and okex, but on a few exchanges like kucoinfutures, you can set multiple trades on the same market for different leverages and margin modes.